### PR TITLE
Fix #493: Publish Inbox API for Mobile Token

### LINKED ISSUE #493: Publish Inbox API for Mobile Token
Publish an Inbox API for Mobile Token as a facade for Push Server REST API with following endpoints:
- POST `/api/inbox/count` - get count of unread messages in inbox
- POST `/api/inbox/message/list` - get a page with messages with given page number and size
- POST `/api/inbox/message/detail` - get detail of a message with given message identifier
- POST `/api/inbox/message/read` - mark a message with given message identifier as read
- POST `/api/inbox/message/read-all` - mark all messages as read for user

The reason of using `POST` for all these endpoints is that we do not want to expose any sensitive details in the URL and for consistency with e.g. POST `/api/auth/token/app/operation/list`.

### PR DESCRIPTION
**Pull request by romanstrobl**

### CODE REVIEW

File: enrollment-server-api-model/pom.xml [hunk 1]
```diff
@@ -33,6 +33,24 @@
         <version>1.4.0-SNAPSHOT</version>
     </parent>
 
+    <dependencies>
+        <dependency>
+            <groupId>jakarta.validation</groupId>
+            <artifactId>jakarta.validation-api</artifactId>
+            <version>${jakarta-validation-api.version}</version>
+            <scope>compile</scope>
```
Comment (banterCZ) [inline enrollment-server-api-model/pom.xml:41]: Default. Any reason to declare it explicitly?

File: enrollment-server/src/main/java/com/wultra/app/enrollmentserver/controller/api/InboxController.java [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+ * Controller with facade for Inbox services in Push server.
+ *
+ * @author Roman Strobl, roman.strobl@wultra.com
+ */
+@ConditionalOnExpression("!'${powerauth.push.service.url}'.empty and ${enrollment-server.inbox.enabled}")
+@RestController
+@RequestMapping(value = "api/inbox")
+public class InboxController {
+
+    private static final Logger logger = LoggerFactory.getLogger(InboxController.class);
```
Comment (banterCZ) [inline enrollment-server/src/main/java/com/wultra/app/enrollmentserver/controller/api/InboxController.java:56]: `@Slf4j`?

**Review by banterCZ — APPROVED** [review]
Looks good.

File: enrollment-server-api-model/pom.xml (hunk 1 above)
Comment (romanstrobl) [reply enrollment-server-api-model/pom.xml:41]: Just a copy-paste reason.

File: enrollment-server/src/main/java/com/wultra/app/enrollmentserver/controller/api/InboxController.java (hunk 2 above)
Comment (romanstrobl) [reply enrollment-server/src/main/java/com/wultra/app/enrollmentserver/controller/api/InboxController.java:56]: Yes.

**Pull request merged**